src/components/common/Problem.vue

```diff
@@ -29,28 +30,100 @@ export default defineComponent({
 		const _file: Ref<string> = ref(props.file);
 		const renderDiv = ref<HTMLElement>();
 
+		// modified from https://github.com/tserkov/vue-plugin-load-script
+		async function loadScript(src: string) {
+			console.log('loading scripts...', src);
+			return new Promise<void>(function (resolve, reject) {
+				let shouldAppend = false;
+				let el: HTMLScriptElement | HTMLLinkElement | null;
+				create_el: if (/\.js(?:\??[0-9a-zA-Z=]*)$/.exec(src)) {
+					el = document.querySelector('script[src="' + src + '"]');
+					if (el?.hasAttribute('data-loaded')) {
+						resolve();
+						console.log(`${src} seems to have already been loaded!`);
+						return;
+					} else if (el) {
+						break create_el;
+					}
+
+					el = document.createElement('script');
+					el.type = 'text/javascript';
+					el.defer = true;
+					el.src = src;
+					shouldAppend = true;
+				} else if (/\.css(?:\??[0-9a-zA-Z=]*)$/.exec(src)) {
+					el = document.querySelector('link[href="' + src + '"]');
+					if (el && el.hasAttribute('data-loaded')) {
```

Comment:
I guess I should have pointed out this instance that is the same as line 41.  I assumed you would catch both of them.  Bad assumption on my part.

Reply:
Apparently I got too excited to check off all the feedback... 😂